tests/codegen/nullability-tests/build.gradle.kts

```diff
@@ -23,6 +23,10 @@ tasks.generateSmithyProjections {
     smithyBuildConfigs.set(files("smithy-build.json"))
 }
 
+tasks.kotlinSourcesJar {
+    dependsOn(tasks.generateSmithyProjections)
+}
```

Comment:
question: You said these `dependsOn` additions were required to fix your `ShapeExt.kt:35:80 Unresolved reference.` errors? That seems odd...

Reply:
The `ShapeExt.kt:35:80 Unresolved reference` errors were fixed by: https://github.com/smithy-lang/smithy-kotlin/blob/423dc01680127b6552f140716aa3429b64edc1af/.github/workflows/artifact-size-metrics.yml#L21-L26

These `dependsOn` were fixes for these types of errors that I started getting afterwards:
```
A problem was found with the configuration of task ':tests:codegen:nullability-tests:generateSmithyProjections' (type 'SmithyBuildTask').
  - Gradle detected a problem with the following location: '/.../smithy-kotlin/tests/codegen/nullability-tests/build/smithyprojections/nullability-tests'.
    
    Reason: Task ':tests:codegen:nullability-tests:kotlinSourcesJar' uses this output of task ':tests:codegen:nullability-tests:generateSmithyProjections' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
```

Reply:
Ah ok. Was that a warning or an error? We've had them for a long time if I recall correctly but they were never errors. Maybe on a newer version of Gradle?

Reply:
Yeah, they were errors. I'm not exactly sure what would changed. I was getting them in my machine as well